src/LaunchDarkly.ServerSdk.Redis/RedisDataStoreBuilder.cs

```diff
@@ -0,0 +1,226 @@
+        /// <returns>the builder</returns>
+        public RedisDataStoreBuilder ConnectTimeout(TimeSpan timeout)
+        {
+            _redisConfig.ConnectTimeout = (int)timeout.TotalMilliseconds;
+            return this;
+        }
+
+        /// <summary>
+        /// Specifies the maximum time to wait for data on the Redis connection.
+        /// </summary>
+        /// <param name="timeout">the timeout interval</param>
+        /// <returns>the builder</returns>
+        /// <seealso cref="OperationTimeout(TimeSpan)"/>
+        public RedisDataStoreBuilder ResponseTimeout(TimeSpan timeout)
+        {
+            _redisConfig.ResponseTimeout = (int)timeout.TotalMilliseconds;
+            return this;
+        }
+
+        /// <summary>
+        /// Specifies the maximum time to wait for each synchronous Redis operation to complete.
+        /// If you are seeing timeout errors - which could result from either an overburdened
+        /// Redis server, or an unusually large operation such as storing a very large feature
+        /// flag - you may want to increase this setting.
+        /// </summary>
+        /// <param name="timeout">the timeout interval</param>
+        /// <returns>the builder</returns>
+        /// <seealso cref="ResponseTimeout(TimeSpan)"/>
+        public RedisDataStoreBuilder OperationTimeout(TimeSpan timeout)
+        {
+            _redisConfig.SyncTimeout = (int)timeout.TotalMilliseconds;
+            return this;
+        }
+
+        /// <summary>
+        /// Specifies the namespace prefix for all keys stored in Redis.
+        /// </summary>
+        /// <param name="prefix">the namespace prefix, or null to use <see cref="Redis.DefaultPrefix"/></param>
+        /// <returns>the builder</returns>
+        public RedisDataStoreBuilder Prefix(string prefix)
+        {
+            _prefix = string.IsNullOrEmpty(prefix) ? Redis.DefaultPrefix : prefix;
+            return this;
+        }
+
+        /// <summary>
+        /// Allows you to modify any of the configuration options supported by StackExchange.Redis
+        /// directly. The current configuration will be passed to your Action, which can modify it
+        /// in any way.
+        /// </summary>
+        /// <example>
+        /// <code>
+        ///     Redis.DataStore()
+        ///         .RedisConfigChanges((config) => {
+        ///             config.Ssl = true;
+        ///         })
+        /// </code>
+        /// </example>
+        /// <param name="modifyConfig"></param>
+        /// <returns></returns>
+        public RedisDataStoreBuilder RedisConfigChanges(Action<ConfigurationOptions> modifyConfig)
+        {
+            modifyConfig.Invoke(_redisConfig);
+            return this;
+        }
+
+        /// <inheritdoc/>
+        public IPersistentDataStore CreatePersistentDataStore(LdClientContext context) =>
+            new RedisDataStoreImpl(_redisConfig, _prefix, context.Basic.Logger.SubLogger("DataStore.Redis"));

```

Comment:
The call to `SubLogger` here is for the new [standard system of logger names](https://github.com/launchdarkly/dotnet-server-sdk-private/blob/6.0/src/LaunchDarkly.ServerSdk/LogNames.cs) that's being introduced in SDK 6.0.

The SDK's base logger name used to be the same as its full class name, `LaunchDarkly.Client.LdClient`— but the loggers for individual components were using their own fairly arbitrary class names, like `LaunchDarkly.Client.Redis.RedisFeatureStoreCore`.

In the new system, the SDK's base logger name is `LaunchDarkly.Sdk.Server.LdClient`— which still corresponds to its class name— but subcomponents use standardized suffixes instead of class names. So, everything related to data stores will be under `LaunchDarkly.Sdk.Server.LdClient.DataStore`, and in this case we're adding a further suffix of `.Redis`. Using hierarchical dot-delimited names is standard in frameworks like log4net (or, in Java, SLF4J), so that you can for instance define a default log filter that applies to everything with the prefix `A.B.C` but also a more specific filter for `A.B.C.D.E`.